src/radical/pilot/agent/lm/prte.py

```diff
@@ -0,0 +1,284 @@
+        # Additional (debug) arguments to prte
+        if os.environ.get('RADICAL_PILOT_PRUN_VERBOSE'):
+            debug_strings = [
+                             '--debug-devel',
+                             '--mca odls_base_verbose 100',
+                             '--mca rml_base_verbose 100'
+                            ]
+        else:
+            debug_strings = []
+
+        # Split up the debug strings into args and add them to the cmdline
+        cmdline += ' '.join(debug_strings)
+
+        vm_size = len(lrms.node_list)
+        logger.info("Start prte on %d nodes ['%s']", vm_size, ' '.join(cmdline))
+        profiler.prof(event='dvm_start', uid=cfg['pilot_id'])
+
+        dvm_uri     = None
+        dvm_process = mp.Popen(cmdline.split(), stdout=mp.PIPE, stderr=mp.STDOUT)
+
+        while True:
+
+            line = dvm_process.stdout.readline()
+
+            if '://' in line:
+
+                dvm_uri = line.strip()
+                logger.info("prte uri: %s" % dvm_uri)
+
+            elif 'DVM ready' in line:
+                if not dvm_uri:
+                    raise Exception("VMURI not found!")
+
+                logger.info("prte startup successful!")
+                profiler.prof(event='dvm_ok', uid=cfg['pilot_id'])
+                break
+
+            else:
+
+                # Check if the process is still around,
+                # and log output in debug mode.
+                if dvm_process.poll() is None:
+                    logger.debug("PRUN: %s", line)
+                else:
+                    # Process is gone: fatal!
+                    raise Exception("prun process disappeared")
+                    profiler.prof(event='dvm_fail', uid=cfg['pilot_id'])
+
+
+        # ----------------------------------------------------------------------
+        def _watch_dvm():
+
+            logger.info('starting prte watcher')
+
+            retval = dvm_process.poll()
+            while retval is None:
+                line = dvm_process.stdout.readline().strip()
+                if line:
+                    logger.debug('prte output: %s', line)
+                else:
+                    time.sleep(1.0)
+
+            if retval != 0:
+                # send a kill signal to the main thread.
+                # We know that Python and threading are likely not to play well
+                # with signals - but this is an exceptional case, and not part
+                # of the stadard termination sequence.  If the signal is
+                # swallowed, the next `prun` call will trigger
+                # termination anyway.
+                os.kill(os.getpid())
+
+            logger.info('prte stopped (%d)' % dvm_process.returncode)
+        # ----------------------------------------------------------------------
+
+        dvm_watcher = ru.Thread(target=_watch_dvm, name="DVMWatcher")
+        dvm_watcher.start()
+
+        lm_info = {'dvm_uri'     : dvm_uri,
+                   'version_info': {name: prte_info}}
```

Comment:
Isn't a `self.` missing from here?

Reply:
where?  But since this is a classmethod, we have no `self` in the first place.

Reply:
Here: ` 'version_info': {name: prte_info}` I was not able to find where name was set up in the method. That is why I asked.

Reply:
@andre-merzky, if this is my mistake go forward and merge!

Reply:
Ah, I see - `name` came from line 35 in this case:
```
    @classmethod
    def lrms_config_hook(cls, name, cfg, lrms, logger, profiler):
```

But looking over it again, we don't actually need to use the name here, the `lm_info` dict is private anyway, so I simplified the dict.  Since that has no functional consequences (the dict entry is informative only), I'll go ahead and merge.

Thanks for your patience! :-)

Reply:
:D Then it is done!